src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +489,106 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
+                        var state = entry[nameof(LogHistoryColumn.State)].ToString();
+
+                        if (type == commandType && state == "Committed" &&
+                            DeleteDatabaseCommandGuidIsValid(guid, out var nodes) &&
+                            nodes.Contains(serverStore.NodeTag))
+                        {
+                            if (entry[nameof(LogHistoryColumn.ExceptionType)] != null)
+                            {
+                                if (index == null)
+                                    hasExceptions = true;
+                                continue; // has errors after apply
+                            }
+
+                            hasExceptions = false;
+                            index = curIndex;
+                        }
+                    }
+                }
+            }
+            return index;
+        }
+
+        private static bool DeleteDatabaseCommandGuidIsValid(string guid, out List<string> nodes)
+        {
+            /*the DeleteDatabaseCommand command id should look like that: "db1/A_B_C/a6m8me9843jt"
+                ($"{databaseName}/{string.Join('_', nodes}")/{newGuid}*/
+
+            nodes = null;
+
+            int index1 = guid.IndexOf('/');
+            if (index1 < 0)
+                return false;
+
+            try
+            {
+
+                index1++;
+                var index2 = guid.IndexOf('/', index1);
+                if (index2 < 0)
+                    return false;
+
+                nodes = guid.Substring(index1, index2 - index1).Split('_').Where(x => x != string.Empty).ToList();
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static bool TryGetLastCommittedDeleteRaftIndexByGuid(ServerStore serverStore, string guid, out long index, out DynamicJsonValue cmd)
+        {
+            index = 0L;
+            cmd = null;
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+                using var _ = Slice.From(context.Allocator, guid, out var guidSlice);
+
+                foreach (var entryHolder in table.SeekByPrimaryKey(guidSlice, 0))
+                {
+                    var entry = ReadHistoryLog(context, entryHolder);
+
+                    var state = entry[nameof(LogHistoryColumn.State)].ToString();
+                    if (state != "Committed")
+                        continue;
+                    var cmdIndex = long.Parse(entry[nameof(LogHistoryColumn.Index)].ToString());
+                    var guidFromCmd = entry[nameof(LogHistoryColumn.Guid)].ToString();
+                    if (guid == guidFromCmd)
```

Comment:
this is strange... you search by prefix, but compare the entire string?

Reply:
Done